src/cloud-api/src/client/region.rs

```diff
@@ -8,49 +8,181 @@
 // by the Apache License, Version 2.0.
 
 //! This module implements Materialize cloud API methods
-//! to retrieve a region's information.
+//! to GET, CREATE or DELETE a region.
+//! To delete an region correctly make sure to
+//! contact support.
 //!
-//! The main usage for a region is only to retrieve the
-//! environment controller and acknowledge if the
-//! region is enabled.
+//! For a better experience retrieving all the available
+//! environments, use [`Client::get_all_regions()`]
 
-use reqwest::Method;
-use serde::Deserialize;
-use url::Url;
+use std::time::Duration;
+
+use chrono::{DateTime, Utc};
+use reqwest::{Method, StatusCode};
+use serde::{Deserialize, Deserializer, Serialize};
 
 use crate::client::cloud_provider::CloudProvider;
 use crate::client::{Client, Error};
 
-/// Represents the region
+/// A customer region is represented in this structure
 #[derive(Debug, Deserialize, Clone)]
 #[serde(rename_all = "camelCase")]
 pub struct Region {
-    /// Represents the cluster name:
+    ///  may not be set if the region is in the process
+    /// may not be set if the region is in the process
+    /// of being created
+    pub region_info: Option<RegionInfo>,
+}
+
+/// Connection details for an active region
+#[derive(Debug, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct RegionInfo {
+    /// Represents the environmentd PG wire protocol address.
     ///
-    /// E.g.: `mzcloud-production-eu-west-1-0`
-    pub cluster: String,
-    /// Represents the complete environment controller url.
+    /// E.g.: 3es24sg5rghjku7josdcs5jd7.eu-west-1.aws.materialize.cloud:6875
+    pub sql_address: String,
+    /// Represents the environmentd HTTP address.
     ///
-    /// E.g.: `https://ec.0.eu-west-1.aws.cloud.materialize.com:443`
-    pub environment_controller_url: Url,
+    /// E.g.: 3es24sg5rghjku7josdcs5jd7.eu-west-1.aws.materialize.cloud:443
+    pub http_address: String,
+    /// Indicates true if the address is resolvable by DNS.
+    pub resolvable: bool,
+    /// The time at which the region was enabled
+    pub enabled_at: Option<DateTime<Utc>>,
 }
 
 impl Client {
```

Comment:
these methods are mostly copies of the removed ones in `environment.rs`, but the diff doesn't show that too well